frontend/src/services/api/ApiService.ts

```diff
@@ -0,0 +1,65 @@
+import type { Endpoint, EndpointReq, EndpointRes } from "./endpoints/endpoint";
+
+const BASE_URL = import.meta.env.VITE_BACKEND_URL;
+
+class ApiService {
+    /**
+     * Perform a fetch to the provided endpoint, with one re-attempt if initial call fails.
+     * 
+     * @param endpoint 
+     * @param payload 
+     * @param retry 
+     * @returns 
+     */
+    public async call<E extends Endpoint<any, any>>(
+        endpoint: E,
+        payload?: EndpointReq<E>,
+        accessToken?: string | null,
+    ): Promise<EndpointRes<E>> {
+        
+        const url = new URL(`api${endpoint.route}`, BASE_URL);
```

Comment:
```suggestion
        const url = new URL(`api/${endpoint.route}`, BASE_URL);
```

Looks minor but it should be safer to have the initial forward slash here and have the convention for `Endpoint.route` to not have an initial slash.

This guards against a possible mistake where the dev forgets where the forward slash is expected.

If `Endpoint.route` is expected to have the initial forward slash, which is the current implementation, accidentally not adding it would be bad since we'd end up with a concatenated URI without a separator (e.g. `apiauth/guest`)

On the other hand, if `Endpoint.route` is expected to **not** have the initial forward slash, accidentally adding it would most likely be fine since most routers automatically truncate duplicated forward slashes.

More on this here: https://stackoverflow.com/questions/10161177/url-with-multiple-forward-slashes-does-it-break-anything